crates/common/mqtt_channel/src/config.rs

```diff
@@ -41,6 +43,11 @@ pub struct Config {
     ///
     /// Default: `1024 * 1024`.
     pub max_packet_size: usize,
+
+    /// LastWill message for a mqtt client
+    ///
+    /// Default: None
+    pub last_will_message: Option<LastWill>,
```

Comment:
The last will message is a regular message. Furthermore, it's better to avoid to expose `rumqttc` types.

```suggestion
    pub last_will_message: Option<Message>,
```

Reply:
fixed